modules/administration-guide/partials/proc_migration-to-devworkspace-engine.adoc

```diff
@@ -0,0 +1,209 @@
+
+[id="migration-to-devworkspace-engine_{context}"]
+= Migration to {devworkspace} engine.
+
+This procedure describes how to migration to {devworkspace} engine using https://docs.openshift.com/container-platform/latest/operators/understanding/olm/olm-understanding-olm.html[OLM] to support the Devfile 2.0.0 file format and mentions how to do so on existing instances.
+
+.Prerequisites
+
+* The `{orch-cli}` tool is available.
+* An instance of {prod-short} deployed using xref:installation-guide:installing-che-on-openshift-4-using-operatorhub.adoc[Operator Hub] from `{prod-channel}` channel on OpenShift cluster version greater or equal to 4.8
+* OpenShift OAuth is enabled. See xref:configuring-openshift-oauth.adoc[].
+* Bundled PostgreSQL
+
+.Procedure
+
+. All workspaces must be stopped and changes pushed back to Git repositories.
+
+. Backup {prod-short} data. See xref:managing-backups-using-chectl.adoc[].
+
+include::partial$snip_scaling-down-che.adoc[]
+
+. Create the script to get all {prod-short} users:
++
+[source,shell,subs="+attributes"]
+----
+cat >get-all-{prod-id}-users.sh<<EOF
+#!/bin/bash
+set -e
+
+ALL_USERS={prod-id}-users.txt
+IDENTITY_PROVIDER_URL=\$({orch-cli} get checluster {prod-checluster} -n {prod-namespace} -o jsonpath="{.status.keycloakURL}" )
+IDENTITY_PROVIDER_SECRET=\$({orch-cli} get checluster/{prod-checluster} -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderSecret}")
+IDENTITY_PROVIDER_PASSWORD=\$(if [ -z "\$IDENTITY_PROVIDER_SECRET" ] || [ \$IDENTITY_PROVIDER_SECRET = "null" ]; then {orch-cli} get checluster/{prod-checluster}  -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderPassword}"; else {orch-cli} get secret \$IDENTITY_PROVIDER_SECRET -n {prod-namespace} -o jsonpath="{.data.password}" | base64 -d; fi)
+IDENTITY_PROVIDER_USERNAME=\$(if [ -z "\$IDENTITY_PROVIDER_SECRET" ] || [ \$IDENTITY_PROVIDER_SECRET = "null" ]; then {orch-cli} get checluster/{prod-checluster}  -n {prod-namespace} -o jsonpath="{.spec.auth.IdentityProviderAdminUserName}"; else {orch-cli} get secret \$IDENTITY_PROVIDER_SECRET -n {prod-namespace} -o jsonpath="{.data.user}" | base64 -d; fi)
+IDENTITY_PROVIDER_REALM=\$({orch-cli} get checluster {prod-checluster} -n {prod-namespace} -o jsonpath="{.spec.auth.identityProviderRealm}")
```

Comment:
An observation is that the scripts would work only if Eclipse Che / CRW is installed in the recomended namespace e.g. `eclipse-che` for upstream. However, there is possibility for installing operator in different namespaces e.g. dogfooding on che-dev cluster